domains/runtime/evm/src/lib.rs

```diff
@@ -856,11 +863,35 @@ impl_runtime_apis! {
         }
 
         fn check_transaction_validity(
-            _uxt: &<Block as BlockT>::Extrinsic,
-            _block_hash: <Block as BlockT>::Hash,
+            uxt: &<Block as BlockT>::Extrinsic,
+            block_hash: <Block as BlockT>::Hash,
         ) -> Result<(), domain_runtime_primitives::CheckTxValidityError> {
-            // TODO: check transaction fee to core-evm
-            Ok(())
+            let lookup = frame_system::ChainContext::<Runtime>::default();
+            let maybe_signer_info = extract_signer_inner(uxt, &lookup);
+
+            if let Some(signer_info) = maybe_signer_info {
+                let signer = signer_info.map_err(|tx_validity_error| {
+                    domain_runtime_primitives::CheckTxValidityError::InvalidTransaction {
+                        error: tx_validity_error,
+                        storage_keys: vec![],
+                    }
+                })?;
```

Comment:
The `storage_keys` in `CheckTxValidityError::InvalidTransaction` is used to construct the fraud proof, can't see why we would return an empty value here. How about replacing `CheckTxValidityError::Lookup` with `CheckTxValidityError::ExtractSigner` and returning it instead?

Reply:
The reason I was returning empty vector array was because at that point, no storage keys were accessed. But, yes, the `CheckTxValidityError::Lookup` seems to be more natural choice.

Reply:
No, what I suggest is "How about replacing CheckTxValidityError::Lookup with CheckTxValidityError::ExtractSigner and returning it instead?" because not just lookup could cause error.

Reply:
Umm, there is no `CheckTxValidityError::Lookup`. In any case better idea would be to have specific error such as `CheckTxValidityError::InvalidSigner`. As the error name should indicate what is the error and not just at what step the error has occurred.